lib/ravelin/supplier.rb

```diff
@@ -0,0 +1,30 @@
+module Ravelin
+  class Supplier < RavelinObject
+    attr_accessor :supplier_id,
+      :type,
+      :group_id,
+      :group_name,
+      :registration_time,
+      :email,
+      :email_verified_time,
+      :name,
+      :telephone,
+      :telephone_verified_time,
+      :telephone_country,
+      :home_location,
+      :market,
+      :market_country,
+      :level,
+      :employment_type,
+      :transport_type,
+      :category,
+      :tags,
+      :custom
+
+    attr_required :supplier_id, :type
+
+    def home_location(obj)
+      @home_location = Ravelin::Location.new(obj)
+    end
```

Comment:
This will overwrite the `:home_location` accessor. You're essentially doing: 
```ruby
module Ravelin
  class Supplier < RavelinObject
    def home_location=(value)
      @home_location = value
    end
    def home_location
      @home_location
    end
    def home_location(obj)
      @home_location = Ravelin::Location.new(obj)
    end
  end
end
```
This means that the read method will be replaced with another that requires one parameter. We'll never be able to read the value without setting it as well.
```
dev(main)> supplier =Ravelin::Supplier.new
=> #<Ravelin::Supplier:0x00007fe6f6781448>
dev(main)> x.home_location = true
=> true
dev(main)> x.home_location
ArgumentError: wrong number of arguments (given 0, expected 1)
from (pry):9:in `home_location'
dev(main)> x.home_location false
=> false
```
This is probably not the intended effect.

Reply:
Are all of these wrong in the same way: https://github.com/deliveroo/ravelin-ruby/blob/master/lib/ravelin/order.rb#L28-L38 or have I screwed up somehow?

Reply:
You have 😄 
You forgot a `=` in the method name:
```ruby
    def home_location=(obj)
      @home_location = Ravelin::Location.new(obj)
    end
```